src/components/Main.jsx

```diff
@@ -7,9 +7,10 @@ import UploadSnippet from './UploadSnippet';
 import About from './About';
 import SignIn from './SignIn';
 
+import '../styles/main/main.styl';
 
 const Main = () => (
-  <main>
+  <main className="main">
     <Switch>
```

Comment:
Just a question: why does switch-route is in main component, and not in application component? Is that front-end best practices? I mean I'm curious, I thought any switch-route should be as high as possible... like in App component. But I may be wrong..

Reply:
As far as header and sidebar should be static and not re-render after some state changes inside, I think it's a better way to organize routing inside Main component since dynamic part of application is inside main